infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -552,6 +562,30 @@ private void handleTxToken(CorfuPayloadMsg<TokenRequest> msg, ChannelHandlerCont
         handleAllocation(msg, ctx, r);
     }
 
+    /**
+     * this method serves a monotonically increasing counter as an Ordered GUID.
+     * It simply returns a counter along with the current epoch value.
+     * If the sequencer restarts the epoch ticks up while the counter resets, thereby
+     * allowing a globally unique monotonically increasing sequence to be served.
+     *
+     * Not Thread Safe: Only OK when SequencerServer is single threaded.
+     *
+     * @param msg corfu message containing guid request
+     * @param ctx netty ChannelHandlerContext
+     * @param r   server router
+     */
+    private void handleOrderedGuidToken(CorfuPayloadMsg<TokenRequest> msg,
+                                        ChannelHandlerContext ctx, IServerRouter r) {
+        final TokenRequest req = msg.getPayload();
+
+        // GlobalSequence points to the next available value, so return it first...
+        Token token = new Token(sequencerEpoch, globalSequence);
+        // ... then increment it by the number of such tokens requested.
+        globalSequence += req.getNumTokens();
```

Comment:
It seems not making sense to increase the sequence by `numTokens`  and return only one sequence. Besides the requested `numTokens` in SequencerView is always 1. This seems weird, right?

Reply:
2 reasons:
I am following same patter as next() Token, so it is copy pasted convention in a way.
There was a discussion that we will soon need ability to request multiple tokens/ids in one request atomically. So this just provisions for that.